packages/cli/src/utils/verifyManifest.ts

```diff
@@ -18,11 +18,27 @@ const logoLibraryFilenameExtension = z.enum(['svg', 'jpg', 'jpeg', 'ai', 'eps',
 
 const appType = z.enum(['content-block', 'platform-app', 'theme']);
 
+const secretKeySet = new Set();
+export function resetSecretKeySet() {
+    secretKeySet.clear();
+}
+
 const secretSchema = object({
     label: string(),
-    key: string().refine((value) => /^[\w-]+$/.test(value), {
-        message: "The key should only contain letters from a-z, A-Z, numbers from 0-9, '-' and '_' without any spaces",
-    }),
+    key: string().refine(
+        (key) => {
+            if (secretKeySet.has(key)) {
+                return false;
+            }
+
+            secretKeySet.add(key);
+            return /^[\w-]+$/.test(key);
+        },
+        {
+            message:
+                "The key should only contain letters from a-z, A-Z, numbers from 0-9, '-' and '_' without any spaces",
```

Comment:
Does it make sense to add "key must be unique" to this error message?

Reply:
I agree with @fulopdaniel , it would be nice to add it to the message.

Reply:
great catch updated